### ISSUE TITLE
DOC: warm_start not in API reference

### ISSUE THREAD

**Issue by keithbriggs** [cvxpy/cvxpy#1716]
https://www.cvxpy.org/_modules/cvxpy/problems/problem.html does not mention the `warm_start` argument to `.solve()`.

**Comment by pat749** [comment]
yes i also  saw it

**Comment by rileyjmurray** [comment]
I agree that warm_start should be mentioned (at least indirectly) in the docstring for ``.solve()``.  I think the best way forward here is to reorganize our [User Guide](https://www.cvxpy.org/tutorial/index.html) to give "calling a solver" a top-level section, and have that section contain content from
* https://www.cvxpy.org/tutorial/advanced/index.html#solve-method-options,
*  https://www.cvxpy.org/tutorial/advanced/index.html#setting-solver-options, and
* https://www.cvxpy.org/tutorial/advanced/index.html#warm-start.

Then the docstring of ``prob.solve()`` could include a URL that links to that portion of the User Guide. I think this approach is preferable to documenting warm_start directly since the meaning of that argument can't be easily summarized. It's also not the only important argument to ``.solve()`` which is missing from its docstring.
